src/main/java/algorithms/curated170/medium/minimumknightmoves/MinimumKnightMoves.java

```diff
@@ -0,0 +1,29 @@
+package algorithms.curated170.medium.minimumknightmoves;
+
+public class MinimumKnightMoves {
+
+    public int minKnightMoves(int x, int y) {
+
+        x = Math.abs(x);
+        y = Math.abs(y);
+
+        if (x < y) {
+            int temp = x;
+            x = y;
+            y = temp;
+        }
+        if (x == 1 && y == 0) {
+            return 3;
+        }
+        if (x == 2 && y == 2) {
+            return 4;
+        }
+        int delta = x - y;
+        if (2 * y > x) {
+            return (int) (delta - 2 * Math.floor((float) (delta - y) / 3));
+        } else {
+            return (int) (delta - 2 * Math.floor((float) (delta - y) / 4));
+        }
+    }
```

Comment:
I think, this is really a good explanation how we derive this formula:

![image](https://user-images.githubusercontent.com/60903744/124367751-b6665e80-dc62-11eb-82c2-9175ef5e9ad6.png)

https://stackoverflow.com/questions/2339101/knights-shortest-path-on-chessboard/41704071#comment74156588_41704071